test/integration/vhds.h

```diff
@@ -0,0 +1,292 @@
+#include "envoy/config/route/v3/route.pb.h"
```

Comment:
It's not obvious how moving all this code to headers improves the split of core/extension tests, can you comment on this in either the review or commit message?

Reply:
It's not about moving it to headers.
The issue is that the on demand filter is required for core tests to pass.  The tests using on-demand filter share a bunch of code with core tests, so I moved the test definitions to their own files so they could share code rather than me copy-pasting hds test set up.  I can split it into vds.h and vds.cc if you prefer